java/src/PaginationExample.java

```diff
@@ -0,0 +1,154 @@
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * This example shows how pagination works in the FLOW api by printing out all
+ * survey instances ids for a given survey
+ * 
+ * usage: see pagination-example shell script
+ */
+public class PaginationExample {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private final String host;
+    private final String secret;
+    private final String accessKey;
+    private final String surveyId;
+
+    @SuppressWarnings("unchecked")
+    public PaginationExample(String[] args) throws ParseException,
+            InvalidKeyException, NoSuchAlgorithmException,
+            ClientProtocolException, IOException {
+        CommandLine commandLine = new BasicParser().parse(
+                getCommandLineOptions(), args, true);
+
+        host = commandLine.getOptionValue("host");
+        secret = commandLine.getOptionValue("secret");
+        accessKey = commandLine.getOptionValue("access_key");
+        surveyId = commandLine.getOptionValue("survey_id");
+
+        Map<String, Object> first = get(getSurveyInstancesURL());
+        printInstanceIds((List<Map<String, Object>>) first
+                .get("survey_instances"));
+        String since = getSince(first);
+
+        while (since != null) {
+            Map<String, Object> next = get(getSurveyInstancesURL(since));
+            printInstanceIds((List<Map<String, Object>>) next
+                    .get("survey_instances"));
+            since = getSince(next);
+        }
+
+        System.out.println("done.");
+    }
+
+    @SuppressWarnings("unchecked")
+    private String getSince(Map<String, Object> response) {
+        Map<String, Object> meta = (Map<String, Object>) response.get("meta");
+        if (meta == null) {
+            return null;
+        }
+        return (String) meta.get("since");
+    }
+
+    private static void printInstanceIds(
+            List<Map<String, Object>> surveyInstances) {
+        for (Map<String, Object> surveyInstace : surveyInstances) {
+            System.out.println(surveyInstace.get("keyId"));
+        }
+    }
+
+    public String getSurveyInstancesURL() {
+        return host + "/api/v1/survey_instances?surveyId=" + surveyId;
+    }
+
+    public String getSurveyInstancesURL(String since) {
+        return getSurveyInstancesURL() + "&since=" + since;
+    }
+
+    public Map<String, Object> get(String url) throws InvalidKeyException,
+            NoSuchAlgorithmException, ClientProtocolException, IOException {
+        int beginIndex = url.indexOf("/api");
+        int endIndex = url.indexOf("?");
+        endIndex = endIndex == -1 ? url.length() : endIndex;
```

Comment:
Just a stylistic comment but perhaps here its enough to say `url.indexOf("?") == -1` as the ternary condition, then `endIndex` will not have to be assigned to itself again.